packages/react-components/src/components/Drawer/DrawerMenu.tsx

```diff
@@ -104,28 +104,51 @@ export const _DrawerMenu = (props: _DrawerMenuProps): JSX.Element => {
       onLightDismiss={props.onLightDismiss}
       heading={props.heading}
     >
-      <Stack styles={props.styles} role="menu" data-ui-id="drawer-menu">
-        {menuItemsToRender?.slice(0, 1).map((item) => (
-          <DrawerMenuItem
-            {...item}
-            key={`${item.itemKey}` + '0'}
-            shouldFocusOnMount={true}
-            styles={modifiedFirstItemStyle}
-            onItemClick={(ev, itemKey) => {
-              onItemClick(item, ev, itemKey);
-            }}
-          />
-        ))}
-        {menuItemsToRender?.slice(1).map((item, i) => (
-          <DrawerMenuItem
-            {...item}
-            key={`${item.itemKey}` + `${i + 1}`}
-            onItemClick={(ev, itemKey) => {
-              onItemClick(item, ev, itemKey);
-            }}
-          />
-        ))}
-      </Stack>
+      {menuItemsToRender !== undefined && menuItemsToRender[0].itemKey === 'reactions' && (
```

Comment:
Using itemKey to pivot the UI behavior is a little hacky, there is nothing in the interface of `<DrawerMenu />` to suggest that using a specific itemKey will translate to new UI.

For interest of time you can keep this this way, but consider an alternative that would scale to future features that may also want custom behavior as a menu item.

One way would be to expose an onRender function. So `_DrawerMenuItemProps` would have an additional optional property called `onRenderMenuItem` that functions as an override. You can scan the codebase for other onRender functions to get an example of how it works, but conceptually it takes in a react component as the prop and renders that instead. The parent of the DrawerMenu would then pass in a react component that would be rendered instead of a default menu item.

Doing it this way then also keeps the drawer menu interface clean and completely agnostic of the specialized way we use it. (Think, if we ever released DrawerMenu as a pure UI component, most scenarios would not involve Reactions. This is the same reason it doesn't take anything like onHoldClick even though sometimes it has a menu item that performs a hold function)

Reply:
Then the parent component would use like this:
```tsx
<DrawerMenu
    items={[{
      itemKey: 'myKey1',
      onRender: <ReactionDrawerMenuItem />;
    }, {
      //some other item
      itemKey: 'myKey2',
      onClick: () => {}
    }, //etc
   ]}
/>
```

Reply:
Having onRenderMenuItem would again force us to implement similar logic on drawerMenuItem. 
Even if we have a separate reaction drawer menu component, we would still need to decide if the item renders reaction menu item or not.